File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/Java/Go/s283652523_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

const MOD = 1e9 + 7

func solve(n int, a []int) int64 {
	ans := int64(0)
	sum := int64(a[n-1])
	for i := n - 2; i >= 0; i-- {
		term := sum * int64(a[i])
		ans = (ans + term%MOD) % MOD
		sum = (sum + int64(a[i])) % MOD
	}
	return ans % MOD
}

func main() {
	sc := bufio.NewScanner(os.Stdin)
	sc.Split(bufio.ScanWords)
	var n int
	sc.Scan()
	fmt.Sscanf(sc.Text(), "%d", &n)
	a := make([]int, n)
	for i := 0; i < n; i++ {
		sc.Scan()
		fmt.Sscanf(sc.Text(), "%d", &a[i])
	}
	ans := solve(n, a)
	fmt.Println(ans)
}

// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/Java/Python/s283652523_commented.py
import sys

def main():
    data = list(map(int, sys.stdin.read().split()))
    N = data[0]
    A = data[1:N+1]
    MOD = 10**9 + 7
    ans = 0
    sum_val = A[-1]
    for i in range(N-2, -1, -1):
        term = (sum_val * A[i]) % MOD
        ans += term
        ans %= MOD
        sum_val += A[i]
        sum_val %= MOD
    print(ans % MOD)

if __name__ == "__main__":
    main()
# End of Code
